src/application/engine/rendering/ClientMap.java

```diff
@@ -485,4 +489,23 @@ public void setScale(double xFactor, double yFactor)
         gc.setTransform(startingAffine);
         gc.scale(xFactor, yFactor);
     }
+
+    public void writeServerMessage(String readString)
+    {
+        new Thread(() ->
```

Comment:
we should instead have a messagepump with lightevents. on each update we could then iterate its elements and they would remove themselves if they have timedout